packages/core/src/tracing/instrumentation/database/elasticsearchModern.js

```diff
@@ -205,13 +204,46 @@ function onError(span, error) {
     span.data.elasticsearch.error = tracingUtil.getErrorDetails(error);
   }
   if (error.meta && error.meta.meta) {
-    parseConnectionFromResult(span, error.meta);
+    getConnectionDetailsFromResultMeta(span, error.meta);
   }
   span.transmit();
 }
 
-function parseConnectionFromResult(span, result) {
-  // NOTE: Does not work for v8
+function parseAndCacheConnectionUrl(connectionUrl) {
+  if (connectionUrl && connectionUrl instanceof url.URL) {
+    // The connection URL is already in the form of a URL object. No need to parse it.
+    return connectionUrl;
```

Comment:
The parsing result that we put in the cache is a URL object created by calling `new URL(connectionUrl)` and the cache key is the incoming string. The reason for maintaining the cache is so that we do not spend the CPU cycles for parsing the string into a URL object twice.

If the incoming URL parameter is already a URL object, we gain nothing by putting it into the cache. Hope that makes sense?

And the reason for having this initial check whether this is already a URL is that some versions of the library used URL objects in some places and connections strings in other places, IIRC, but that distinction had already be in place before my change.

Reply:
> If the incoming URL parameter is already a URL object, we gain nothing by putting it into the cache. Hope that makes sense?

Ah yes true. Understood 👍